From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 14 Feb 2014 15:56:19 +0100
Subject: man: always place <programlisting> and </programlisting> in a line
 with actual sources,
 so that we don't get spurious newlines in the man page output

(cherry picked from commit 9fccdb0f64d12bc09a71442dd0af2248c1aa3e89)

Conflicts:
	man/systemd-socket-proxyd.xml
---
 man/daemon.xml                           |  3 +--
 man/kernel-install.xml                   |  8 ++------
 man/nss-myhostname.xml                   | 12 ++++--------
 man/sd_journal_get_fd.xml                |  3 +--
 man/systemctl.xml                        |  3 +--
 man/systemd-delta.xml                    | 12 ++++--------
 man/systemd-journal-gatewayd.service.xml | 16 +++++-----------
 man/systemd-run.xml                      |  3 +--
 man/systemd-sleep.conf.xml               |  6 ++----
 man/systemd.service.xml                  | 13 ++++---------
 man/timedatectl.xml                      | 18 ++++++------------
 man/tmpfiles.d.xml                       |  7 ++-----
 12 files changed, 33 insertions(+), 71 deletions(-)

diff --git a/man/daemon.xml b/man/daemon.xml
index 105826a..17d9dd6 100644
--- a/man/daemon.xml
+++ b/man/daemon.xml
@@ -777,8 +777,7 @@ AS_IF([test "x$with_systemdsystemunitdir" = "xyes" -o "x$with_systemdsystemunitd
          [with_systemdsystemunitdir=$def_systemdsystemunitdir])])
 AS_IF([test "x$with_systemdsystemunitdir" != "xno"],
       [AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])])
-AM_CONDITIONAL(HAVE_SYSTEMD, [test "x$with_systemdsystemunitdir" != "xno"])
-</programlisting>
+AM_CONDITIONAL(HAVE_SYSTEMD, [test "x$with_systemdsystemunitdir" != "xno"])</programlisting>
 
                         <para>This snippet allows automatic
                         installation of the unit files on systemd
diff --git a/man/kernel-install.xml b/man/kernel-install.xml
index 929ceef..3612b7d 100644
--- a/man/kernel-install.xml
+++ b/man/kernel-install.xml
@@ -90,9 +90,7 @@ along with systemd; If not, see <http://www.gnu.org/licenses/>.
           <filename>/usr/lib/kernel/install.d/*.install</filename> and
           <filename>/etc/kernel/install.d/*.install</filename> with
           the arguments
-          <programlisting>
-add <replaceable>KERNEL-VERSION</replaceable> <filename>/boot/<replaceable>MACHINE-ID</replaceable>/<replaceable>KERNEL-VERSION</replaceable>/</filename>
-          </programlisting>
+          <programlisting>add <replaceable>KERNEL-VERSION</replaceable> <filename>/boot/<replaceable>MACHINE-ID</replaceable>/<replaceable>KERNEL-VERSION</replaceable>/</filename></programlisting>
           </para>
 
           <para>The kernel-install plugin <filename>50-depmod.install</filename> runs depmod for the <replaceable>KERNEL-VERSION</replaceable>.</para>
@@ -114,9 +112,7 @@ add <replaceable>KERNEL-VERSION</replaceable> <filename>/boot/<replaceable>MACHI
         <listitem>
           <para>Calls every executable <filename>/usr/lib/kernel/install.d/*.install</filename>
           and <filename>/etc/kernel/install.d/*.install</filename> with the arguments
-          <programlisting>
-remove <replaceable>KERNEL-VERSION</replaceable> <filename>/boot/<replaceable>MACHINE-ID</replaceable>/<replaceable>KERNEL-VERSION</replaceable>/</filename>
-          </programlisting>
+          <programlisting>remove <replaceable>KERNEL-VERSION</replaceable> <filename>/boot/<replaceable>MACHINE-ID</replaceable>/<replaceable>KERNEL-VERSION</replaceable>/</filename></programlisting>
           </para>
 
           <para><command>kernel-install</command> removes the entire directory
diff --git a/man/nss-myhostname.xml b/man/nss-myhostname.xml
index efbadac..cca3a99 100644
--- a/man/nss-myhostname.xml
+++ b/man/nss-myhostname.xml
@@ -86,8 +86,7 @@
         <refsect1>
                 <title>Example</title>
 
-                <programlisting>
-# /etc/nsswitch.conf
+<programlisting># /etc/nsswitch.conf
 
 passwd:         compat
 group:          compat
@@ -101,20 +100,17 @@ services:       db files
 ethers:         db files
 rpc:            db files
 
-netgroup:       nis
-                </programlisting>
+netgroup:       nis</programlisting>
 
                 <para>To test, use <command>glibc</command>'s <command>getent</command> tool:</para>
 
-                <programlisting>
-$ getent ahosts `hostname`
+                <programlisting>$ getent ahosts `hostname`
 ::1             STREAM omega
 ::1             DGRAM
 ::1             RAW
 127.0.0.2       STREAM
 127.0.0.2       DGRAM
-127.0.0.2       RAW
-                </programlisting>
+127.0.0.2       RAW</programlisting>
 
                 <para>In this case the local hostname is <varname>omega</varname>.</para>
 
diff --git a/man/sd_journal_get_fd.xml b/man/sd_journal_get_fd.xml
index 4b36f97..df7632f 100644
--- a/man/sd_journal_get_fd.xml
+++ b/man/sd_journal_get_fd.xml
@@ -330,8 +330,7 @@ int wait_for_changes(sd_journal *j) {
         pollfd.events = sd_journal_get_events(j);
         poll(&amp;pollfd, 1, msec);
         return sd_journal_process(j);
-}
-                </programlisting>
+}</programlisting>
         </refsect1>
 
         <refsect1>
diff --git a/man/systemctl.xml b/man/systemctl.xml
index 25b03d8..cde1694 100644
--- a/man/systemctl.xml
+++ b/man/systemctl.xml
@@ -544,8 +544,7 @@ LISTEN           UNIT                        ACTIVATES
 [::]:22          sshd.socket                 sshd.service
 kobject-uevent 1 systemd-udevd-kernel.socket systemd-udevd.service
 
-5 sockets listed.
-            </programlisting>
+5 sockets listed.</programlisting>
             Note: because the addresses might contains spaces, this output
             is not suitable for programmatic consumption.
             </para>
diff --git a/man/systemd-delta.xml b/man/systemd-delta.xml
index ebaa349..f802aa5 100644
--- a/man/systemd-delta.xml
+++ b/man/systemd-delta.xml
@@ -204,20 +204,16 @@
 		<title>Examples</title>
 
 		<para>To see all local configuration:</para>
-		<programlisting>systemd-delta
-		</programlisting>
+		<programlisting>systemd-delta</programlisting>
 
 		<para>To see all runtime configuration:</para>
-		<programlisting>systemd-delta /run
-		</programlisting>
+		<programlisting>systemd-delta /run</programlisting>
 
 		<para>To see all system unit configuration changes:</para>
-		<programlisting>systemd-delta systemd/system
-		</programlisting>
+		<programlisting>systemd-delta systemd/system</programlisting>
 
 		<para>To see all runtime "drop-in" changes for system units:</para>
-		<programlisting>systemd-delta --type=extended /run/systemd/system
-		</programlisting>
+		<programlisting>systemd-delta --type=extended /run/systemd/system</programlisting>
 	</refsect1>
 
         <refsect1>
diff --git a/man/systemd-journal-gatewayd.service.xml b/man/systemd-journal-gatewayd.service.xml
index 0b6fd70..f639811 100644
--- a/man/systemd-journal-gatewayd.service.xml
+++ b/man/systemd-journal-gatewayd.service.xml
@@ -154,14 +154,12 @@ along with systemd; If not, see <http://www.gnu.org/licenses/>.
         <listitem><para>Return a JSON structure describing the machine.</para>
 
         <para>Example:
-        <programlisting>
-{ "machine_id" : "8cf7ed9d451ea194b77a9f118f3dc446",
+        <programlisting>{ "machine_id" : "8cf7ed9d451ea194b77a9f118f3dc446",
   "boot_id" : "3d3c9efaf556496a9b04259ee35df7f7",
   "hostname" : "fedora",
   "os_pretty_name" : "Fedora 19 (Rawhide)",
   "virtualization" : "kvm",
-  ...}
-        </programlisting>
+  ...}</programlisting>
         </para>
         </listitem>
       </varlistentry>
@@ -295,16 +293,12 @@ along with systemd; If not, see <http://www.gnu.org/licenses/>.
     in <ulink
     url="http://www.freedesktop.org/wiki/Software/systemd/export">Journal
     Export Format</ulink>:
-    <programlisting>
-curl --silent -H'Accept: application/vnd.fdo.journal' \
-       'http://localhost:19531/entries?boot'
-    </programlisting>
+    <programlisting>curl --silent -H'Accept: application/vnd.fdo.journal' \
+       'http://localhost:19531/entries?boot'</programlisting>
     </para>
 
     <para>Listen for core dumps:
-    <programlisting>
-curl 'http://localhost:19531/entries?follow&amp;MESSAGE_ID=fc2e22bc6ee647b6b90729ab34a250b1'
-    </programlisting></para>
+    <programlisting>curl 'http://localhost:19531/entries?follow&amp;MESSAGE_ID=fc2e22bc6ee647b6b90729ab34a250b1'</programlisting></para>
   </refsect1>
 
   <refsect1>
diff --git a/man/systemd-run.xml b/man/systemd-run.xml
index 48a0715..97ddf44 100644
--- a/man/systemd-run.xml
+++ b/man/systemd-run.xml
@@ -209,8 +209,7 @@ Sep 08 07:37:21 bupkis systemd[1]: Starting /usr/bin/env...
 Sep 08 07:37:21 bupkis systemd[1]: Started /usr/bin/env.
 Sep 08 07:37:21 bupkis env[19948]: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
 Sep 08 07:37:21 bupkis env[19948]: LANG=en_US.UTF-8
-Sep 08 07:37:21 bupkis env[19948]: BOOT_IMAGE=/vmlinuz-3.11.0-0.rc5.git6.2.fc20.x86_64
-    </programlisting>
+Sep 08 07:37:21 bupkis env[19948]: BOOT_IMAGE=/vmlinuz-3.11.0-0.rc5.git6.2.fc20.x86_64</programlisting>
   </refsect1>
 
   <refsect1>
diff --git a/man/systemd-sleep.conf.xml b/man/systemd-sleep.conf.xml
index dc4b0da..4622205 100644
--- a/man/systemd-sleep.conf.xml
+++ b/man/systemd-sleep.conf.xml
@@ -160,10 +160,8 @@ along with systemd; If not, see <http://www.gnu.org/licenses/>.
     <para>Example: to exploit the <quote>freeze</quote> mode added
     in Linux 3.9, one can use <command>systemctl suspend</command>
     with
-    <programlisting>
-[Sleep]
-SuspendState=freeze
-    </programlisting></para>
+    <programlisting>[Sleep]
+SuspendState=freeze</programlisting></para>
   </refsect1>
 
   <refsect1>
diff --git a/man/systemd.service.xml b/man/systemd.service.xml
index 4fb21ba..e8bb02c 100644
--- a/man/systemd.service.xml
+++ b/man/systemd.service.xml
@@ -442,12 +442,9 @@
                                 be used, they need to be passed
                                 explicitly to a shell implementation
                                 of some kind. Example:</para>
-                                <programlisting>ExecStart=/bin/sh -c 'dmesg | tac'
-                                </programlisting>
-
+                                <programlisting>ExecStart=/bin/sh -c 'dmesg | tac'</programlisting>
                                 <para>Example:</para>
-                                <programlisting>ExecStart=/bin/echo one ; /bin/echo "two two"
-                                </programlisting>
+                                <programlisting>ExecStart=/bin/echo one ; /bin/echo "two two"</programlisting>
                                 <para>This will execute
                                 <command>/bin/echo</command> two
                                 times, each time with one argument,
@@ -460,8 +457,7 @@
 
                                 <para>Example:</para>
                                 <programlisting>ExecStart=/bin/echo / &gt;/dev/null &amp; \; \
-/bin/ls
-                                </programlisting>
+/bin/ls</programlisting>
                                 <para>This will execute
                                 <command>/bin/echo</command> with five
                                 arguments: <literal>/</literal>,
@@ -472,8 +468,7 @@
 
                                 <para>Example:</para>
                                 <programlisting>Environment="ONE=one" 'TWO=two two'
-ExecStart=/bin/echo $ONE $TWO ${TWO}
-                                </programlisting>
+ExecStart=/bin/echo $ONE $TWO ${TWO}</programlisting>
                                 <para>This will execute
                                 <command>/bin/echo</command> with four
                                 arguments: <literal>one</literal>,
diff --git a/man/timedatectl.xml b/man/timedatectl.xml
index f4975ce..5f95486 100644
--- a/man/timedatectl.xml
+++ b/man/timedatectl.xml
@@ -245,8 +245,7 @@
         <refsect1>
                 <title>Examples</title>
                 <para>Show current settings:
-                <programlisting>
-$ timedatectl
+                <programlisting>$ timedatectl
       Local time: Fri, 2012-11-02 09:26:46 CET
   Universal time: Fri, 2012-11-02 08:26:46 UTC
         RTC time: Fri, 2012-11-02 08:26:45
@@ -262,27 +261,22 @@ NTP synchronized: no
  Next DST change: CET → CEST, DST will become active
                   the clock will jump one hour forward
                   Sun, 2013-03-31 01:59:59 CET
-                  Sun, 2013-03-31 03:00:00 CEST
-                </programlisting>
+                  Sun, 2013-03-31 03:00:00 CEST</programlisting>
                 </para>
 
                 <para>Enable an NTP daemon (chronyd):
-                <programlisting>
-$ timedatectl set-ntp true
+                <programlisting>$ timedatectl set-ntp true
 ==== AUTHENTICATING FOR org.freedesktop.timedate1.set-ntp ===
 Authentication is required to control whether network time synchronization shall be enabled.
 Authenticating as: user
 Password: ********
-==== AUTHENTICATION COMPLETE ===
-                </programlisting>
+==== AUTHENTICATION COMPLETE ===</programlisting>
 
-                <programlisting>
-$ systemctl status chronyd.service
+                <programlisting>$ systemctl status chronyd.service
 chronyd.service - NTP client/server
           Loaded: loaded (/usr/lib/systemd/system/chronyd.service; enabled)
           Active: active (running) since Fri, 2012-11-02 09:36:25 CET; 5s ago
-...
-                </programlisting>
+...</programlisting>
                 </para>
         </refsect1>
 
diff --git a/man/tmpfiles.d.xml b/man/tmpfiles.d.xml
index 8267ffc..9578c93 100644
--- a/man/tmpfiles.d.xml
+++ b/man/tmpfiles.d.xml
@@ -109,7 +109,6 @@
 d    /run/user   0755 root root 10d -
 L    /tmp/foobar -    -    -    -   /dev/null</programlisting>
 
-
                 <refsect2>
                         <title>Type</title>
 
@@ -277,13 +276,11 @@ L    /tmp/foobar -    -    -    -   /dev/null</programlisting>
                         </para>
 
                         <para>For example:
-                        <programlisting>
-# Make sure these are created by default so that nobody else can
+                        <programlisting># Make sure these are created by default so that nobody else can
 d /tmp/.X11-unix 1777 root root 10d
 
 # Unlink the X11 lock files
-r! /tmp/.X[0-9]*-lock
-                        </programlisting>
+r! /tmp/.X[0-9]*-lock</programlisting>
                         The second line in contrast to the first one
                         would break a running system, and will only be
                         executed with <option>--boot</option>.</para>
